files/qsub.pl

```diff
@@ -540,7 +540,19 @@ sub parse_script
     # replace PBS_JOBID in -o / -e
     #   torque sets stdout/err in cwd -> so force abspath like done above
     # All script changes here
+    my $change_line = sub {
+        my ($line, $env_var, $prefix) = @_;
+        if ($line =~ m/^\s*#PBS.*?\$\{?$prefix($env_var)\}?/ && $ENV{$1}) {
+            $line =~ s/\$\{?$prefix($env_var)\}?/$ENV{$1}/g;
+        }
+        return ($line);
+    };
+
     foreach my $line (@lines) {
+        $line = &$change_line($line, 'VSC[A-Z_]*', '');
+        $line = &$change_line($line, 'HOME', '');
+        $line = &$change_line($line, 'HOME', 'PBS_O_');
+        $line = &$change_line($line, 'MAIL', 'PBS_O_');
```

Comment:
Maybe also include `$PBS_O_WORKDIR` (which should be replaced with the value of `$PWD`)?

Reply:
Also, can we make this a list you iterate over calling the function?